tests/test_catx.py

```diff
@@ -1,38 +1,96 @@
-from typing import List
+from typing import List, Type, Tuple
 from unittest.mock import MagicMock
 
 import chex
 import jax
 import numpy as np
 import pytest
 import optax
-from chex import ArrayNumpy, Array
+from chex import ArrayNumpy, Array, PRNGKey
 from pytest_mock import MockerFixture
 from jax import numpy as jnp
 
-from catx.catx import CATX
-from catx.network_builder import NetworkBuilder
+from catx.catx import CATX, CATXState
+from catx.network_module import CustomHaikuNetwork
 
-from catx.type_defs import Observations, Actions, Probabilities, JaxObservations
+from catx.type_defs import (
+    Observations,
+    Actions,
+    Probabilities,
+    JaxObservations,
+    JaxProbabilities,
+    JaxActions,
+)
 
 
 @pytest.fixture
-def catx(mlp_builder: NetworkBuilder, request: pytest.FixtureRequest = None) -> CATX:
+def catx(
+    custom_network_without_extras: Type[CustomHaikuNetwork],
+    request: pytest.FixtureRequest = None,
+) -> CATX:
     if not request:
         action_min = 0.0
         action_max = 1.0
     else:
         action_min = request.param[0]
         action_max = request.param[1]
 
-    rng_key = jax.random.PRNGKey(42)
-    rng_key, catx_key = jax.random.split(rng_key, num=2)
+    catx = CATX(
+        custom_network=custom_network_without_extras,
+        optimizer=optax.adam(learning_rate=0.01),
+        discretization_parameter=4,
+        bandwidth=1.5 / 4,
+        action_min=action_min,
+        action_max=action_max,
+    )
+
+    return catx
+
+
+@pytest.fixture
+def catx_init(
+    custom_network_without_extras: Type[CustomHaikuNetwork],
+    jax_observations: JaxObservations,
+    epsilon: float,
+    key: PRNGKey,
+    request: pytest.FixtureRequest = None,
+) -> Tuple[CATX, CATXState]:
+    if not request:
+        action_min = 0.0
+        action_max = 1.0
+    else:
+        action_min = request.param[0]
+        action_max = request.param[1]
 
-    optimizer = optax.adam(learning_rate=0.01)
     catx = CATX(
-        rng_key=catx_key,
-        network_builder=mlp_builder,
-        optimizer=optimizer,
+        custom_network=custom_network_without_extras,
+        optimizer=optax.adam(learning_rate=0.01),
+        discretization_parameter=4,
+        bandwidth=1.5 / 4,
+        action_min=action_min,
+        action_max=action_max,
+    )
+
+    state = catx.init(obs=jax_observations, epsilon=epsilon, key=key)
+
+    return catx, state
```

Comment:
these two fixtures are quite similar. Could `catx_init` actually use the fixture `catx`